mass2/core/misc.py

```diff
@@ -11,67 +15,74 @@
 
 
 def show(fig: plt.Figure | None = None) -> mo.Html:
+    """Create a Marimo interactive view of the given Matplotlib figure (or the current figure if None)."""
     if fig is None:
         fig = plt.gcf()
     return mo.mpl.interactive(fig)
 
 
 def pickle_object(obj: Any, filename: str) -> None:
+    """Pickle the given object to the given filename using dill."""
     with open(filename, "wb") as file:
         dill.dump(obj, file)
 
 
 def unpickle_object(filename: str) -> Any:
+    """Unpickle an object from the given filename using dill."""
     with open(filename, "rb") as file:
         obj = dill.load(file)
         return obj
 
 
 def smallest_positive_real(arr: ArrayLike) -> float:
+    """Return the smallest positive real number in the given array-like object."""
+
     def is_positive_real(x: Any) -> bool:
+        "Is `x` a positive real number?"
         return x > 0 and np.isreal(x)
 
     positive_real_numbers = np.array(list(filter(is_positive_real, np.asarray(arr))))
     return np.min(positive_real_numbers)
 
 
 def good_series(df: pl.DataFrame, col: str, good_expr: pl.Expr, use_expr: bool | pl.Expr) -> pl.Series:
-    # this uses lazy before filting to hopefully allow polars to only access the data needed to filter
-    # and the data needed to output what we want
+    """Return a Series from the given DataFrame column, filtered by the given good_expr and use_expr."""
+    # This uses lazy before filtering. We hope this will allow polars to only access the data needed to filter
+    # and the data needed to output what we want.
     good_df = df.lazy().filter(good_expr)
     if use_expr is not True:
         good_df = good_df.filter(use_expr)
     return good_df.select(pl.col(col)).collect().to_series()
 
 
 def median_absolute_deviation(x: ArrayLike) -> float:
+    """Return the median absolute deviation of the input, unnormalized."""
     x = np.asarray(x)
     return float(np.median(np.abs(x - np.median(x))))
 
 
 def sigma_mad(x: ArrayLike) -> float:
+    """Return the nomrlized median absolute deviation of the input, rescaled to give the rms if distribution is Gaussian."""
     return median_absolute_deviation(x) * 1.4826
 
 
 def outlier_resistant_nsigma_above_mid(x: ArrayLike, nsigma: float = 5) -> float:
+    """Return the value that is nsigma MADs above the median of the input."""
```

Comment:
Return the value that is nsigma median absolute deviations (MADs) above the median of the input.